src/app/space/create/codebases/codebases.component.ts

```diff
@@ -267,7 +286,13 @@ export class CodebasesComponent implements OnDestroy, OnInit {
           this.codebases = cloneDeep(codebases);
           this.codebases.unshift({} as Codebase); // Add empty object for row header
           this.applyFilters(this.appliedFilters);
+        } else {
+          // clear the codebases list:
+          this.allCodebases = [];
+          this.codebases = [];
         }
+        // re-fetch Che state to have consistent information inside all related components:
+        this.fetchCheState();
```

Comment:
I don't believe we want to call fetchCheState() here? The intention was that we broadcast the cheStateChange event only if the Che state changed from idle to running. 

The cheStateChange event now appears to be used to update "related components"? Can you please clarify what components need to be updated?

Perhaps a different event name would be more appropriate, like codebasesUpdate? Then the related components can subscribe to that if necessary.

Reply:
Place number 1: codebases actions: 
https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/codebases-item-actions/codebases-item-actions.component.ts#L20
https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/codebases-item-actions/codebases-item-actions.component.ts#L39

Place number 2: codebase item:
https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/codebases-item/codebases-item.component.ts#L21
https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/codebases-item/codebases-item.component.ts#L47

As long as the codebases list is re-rendered each time on codebases updation - what will happen to those components? they will all be recreated and will have 
`cheRunning: boolean = false;`
and it's not the current actual true state of che. How should they be informed? Where those components have to take that value? Fetch the state on their own?

If user goes to openshift.io and adds a codebase he will not be able to use kebab menu or see the list of workspaces any more. Here is the screenshot:
![screenshot from 2017-11-22 09-44-42](https://user-images.githubusercontent.com/1611939/33116038-e3116fb8-cf6b-11e7-9efb-d4ccc104f0dd.png)

About the "intention was that we broadcast the cheStateChange event only if the Che state changed from idle to running" - the name of the event in such case is not expressive/explicit, by Che state change it could change from idle to running, as well as from running to stopped. I think cheRunning - would avoid such misunderstandings.

About sending "codebasesUpdate" - no need for now cause need the che state. So how it could be solved: 
- maybe place in CheService - the current state of Che - for easy access without additional request to server?

Reply:
I understand why the codebase items are left in a disabled state and realize that we need to fix that. However, the point I'm trying to convey is that the Che state has not changed, it's still running, so broadcasting cheStateChange in this case is misleading.

At least, the name of the cheStateChange event isn't appropriate because we're not using it to reflect a change in state. Instead, we're now using it to force the components to redraw even though Che was already running.

I'll defer to @joshuawilson if we want to merge as is? It works, but could use some refactoring.

Reply:
I agree with @dlabrecq we need to keep the use of the events separate.

Reply:
@dlabrecq will be ok if i pass cheRunning property to components:
`[cheRunning]="cheRunning"`
for example:
`<codebases-item-actions [codebase]="item" [index]="index" [cheRunning]="cheRunning"></codebases-item-actions>`

or you prefer other event to be used?

Reply:
I think that would be ok. Whatever is cleaner?